js/angular-tablesort.js

```diff
@@ -193,22 +376,31 @@ tableSortModule.directive("tsRepeat", ['$compile', function($compile) {
                 noDataRow.removeAttr(tsRepeatDirective);
                 noDataRow.addClass("showIfLast");
                 noDataRow.children().remove();
-                noDataRow.append('<td colspan="' + element[0].childElementCount + '"></td>');
+                noDataRow.append('<td colspan="' + element[0].childElementCount + '">{{noDataText}}</td>');
                 noDataRow = $compile(noDataRow)(scope);
                 element.parent().prepend(noDataRow);
             }
 
+            //pass the `itemsList` from `item in itemsList` to the master directive as a string so it can be used in expressions 
+            tsWrapperCtrl.setDataForPager(repeatInMatch[2])
+
             angular.element(element[0]).attr(ngRepeatDirective, repeatExpr);
             $compile(element, null, 1000000)(scope);
         }
     };
 }]);
 
+tableSortModule.filter( 'tablesortLimit', function(){
+    return function(array, limitFun) {
+       if(!array) return;
+       return limitFun(array);
+    };
+} );
+
 tableSortModule.filter( 'tablesortOrderBy', function(){
     return function(array, sortfun ) {
         if(!array) return;
-        var arrayCopy = [];
-        for ( var i = 0; i < array.length; i++) { arrayCopy.push(array[i]); }
+        var arrayCopy = array.concat();
```

Comment:
This is a slightly faster way to copy an array!  i notices about a 50ms improvement on a table with 10,000 items.